docs/styling-conventions.md

```diff
@@ -1,14 +1,156 @@
 # Component Styling Conventions
 
 ## Summary
-Components must use the [styled-components](https://www.styled-components.com/) package to style all HTML elements they render, and allow certain styles to be specified by a [styled-components theme](https://www.styled-components.com/docs/advanced#theming). Add the default values for any theme properties you use in /src/defaultComponentTheme.js. All theme properties must start with `rui_` followed by a camelcased identifier.
+Components must use the [styled-components](https://www.styled-components.com/) package to style all HTML elements they render, and allow certain styles to be specified by a [styled-components theme](https://www.styled-components.com/docs/advanced#theming).
 
-By convention visual and text styles should be baked into each component. This means that a containing element should set the font-family, font-weight, font-size etc. This convention will make it easier to integrate components into the Reaction Starter Kit and other environments. Further, style values should be retrieved from the theme as seen in the example below.
+## Theme Variables
 
+The default styled-components theme, which is used in the style guide and baked into the components package, is in `/src/defaultComponentTheme.js`.
+
+The following styles should almost always be put into the theme file as variables, rather than hard coded in the component file:
+- padding
+- colors
+- border width
+- border radius
+- font family
+- font size
+- font weight
+
+> ALWAYS use the most specific CSS rule there is. For example, you should never have just a single "Padding" variable, but rather "PaddingLeft", "PaddingRight", "PaddingTop", and "PaddingBottom". Similarly, never use just "Border" or "BorderLeft", but rather the full specific rule such as "BorderLeftColor" or "BorderTopLeftRadius".
+
+In general, flexbox and other layout styles should be baked into the component and not controlled by theme. A component can have props that tell it to use one of several layouts if necessary, or better yet, we can create multiple similar components. Theming should be reserved for borders, colors, padding, and typography.
+
+Height and width should come from theme if the component would still be functional with differing height and width. If the ratio of these two is important to the look or function of the component, then include just a single value in the theme and calculate the other value from that. If the ratio is 1:1, then use a single variable ending in "Size".
```

Comment:
Not clear on why the height and width of a component should be defined in the theme? can you elaborate on this.

Reply:
Not of the component but of something in the component. Two examples we have now are the checkbox height/width and the size of icons or images that appear in the component. I will clarify.